### DISCUSSION TITLE
Can't curl on hooks. Curl not included in docker image?

### DISCUSSION THREAD

**Discussion by OIZid** [bluenviron/mediamtx#3845] [Questions] [answered]
### Question

I've seen multiple examples in the documentation of curl being used with hooks, but when I try to do it, I get:
`2024/10/07 16:00:18 INF [path mystream] runOnReady command exited: exec: "curl": executable file not found in $PATH`

It seems like curl is not installed in the docker image? I'm using `docker run -it --network=host bluenviron/mediamtx:latest-ffmpeg`

Am I missing something?

**Comment by dbason** [comment, marked as answer]
That image is built from alpine which doesn't include curl in its base image.  Including curl can increase the attack surface of a container image so it's normal to not include it in base images.  If you do want to use curl in a medimatx container you'd need to build a new image with it installed.

**Comment by piyushpranjal03** [comment]
For those who need more guidance on how to build the custom image:
Check the docker image I created using debian bookworm-12 as a base image [here](https://hub.docker.com/repository/docker/piyushpranjal/mediamtx-bookworm/general)

Instructions are also added on the docker image page
